1_fetch/src/get_sb_data.R

```diff
@@ -0,0 +1,170 @@
+    tabledl <- rbind(tabledl, tableadd)
+  }
+  count <- 2
+}
+
+#renaming some columns and getting rid of that ugly first column
+colnames(tabledl) <- c('id', 'parent_title', 'title', 'lastUpdated')
+rownames(tabledl) <- NULL
+
+#if you would like to export the table as csv
+filepath <- file.path(outdir, paste0("_sb_dltable.csv"))
+write.csv(tabledl,filepath, row.names = FALSE)
+return(filepath)
+}
+
+downloadchildren <-function(gages, dldir, workdir, outdir, table_sb_dl)
+{
+  #making a copy to work with
+  data_at_gages <- gages
+  
+  itemfails <- "1st"
+  
+  for (row in 1:nrow(table_sb_dl))
+  {
+    item <- table_sb_dl[row, 'id']
+    print(row)
+    print(item)
+    item_file_download(item, dest_dir = dldir, overwrite_file = TRUE)
+    files <- list.files(path = dldir, pattern=NULL, full.names = TRUE)
+    for (file in files)
+    {
+      last3 <- str_sub(file, -3, -1)
+      last3 <- tolower(last3)
+      if (last3 == "xml")
+      {
+        file.remove(file)
+      }
+      if (last3 == "zip")
+      {
+        unzip(file, exdir=workdir)
+        file.remove(file)
+      }
+      if (last3 == "txt")
+      {
+        file.copy(file, workdir)
+        file.remove(file)
+      }
+    }
+    filestomerge <- c(list.files(workdir, ".txt", recursive=TRUE, full.names=TRUE, include.dirs=TRUE),list.files(workdir, ".csv", recursive=TRUE, full.names=TRUE, include.dirs=TRUE),list.files(workdir, ".TXT", recursive=TRUE, full.names=TRUE, include.dirs=TRUE),list.files(workdir, ".CSV", recursive=TRUE, full.names=TRUE, include.dirs=TRUE))
+    for (filem in filestomerge)
+    {
+      print(filem)
+      success = "yes"
+      tryCatch(tempfile <- read.csv(filem, header = TRUE), error = function(e) {success <<- "no"})
+      names(tempfile) <- toupper(names(tempfile))
+      if (!"COMID" %in% colnames(tempfile))
+      {
+        success <- "no"
+      }
+      if (success == "no")
+      {
+        itemfails <- c(itemfails, filem)
+        print("data table error")
+      }
+      if (success == "yes")
+      {
+        print("We're merging")
+        try(tempfile <- subset(tempfile, select =-c(NODATA)), silent = TRUE)
+        try(tempfile <- subset(tempfile, select =-c(CAT_NODATA)), silent = TRUE)
+        try(tempfile <- subset(tempfile, select =-c(ACC_NODATA)), silent = TRUE)
+        try(tempfile <- subset(tempfile, select =-c(TOT_NODATA)), silent = TRUE)
+        data_at_gages <- merge(data_at_gages, tempfile, by = "COMID", all.x = TRUE)
+      }
+      file.remove(filem)
+    }
+    f <- list.files(workdir)
+    unlink(file.path(workdir,f), recursive=TRUE)
+  }
+  
+  if (length(itemfails) > 0)
+  {
+    failist <- as.data.frame(itemfails[-1])
+    colnames(failist) <- c('filename')
+    rownames(failist) <- NULL
+  }
+  filepath1 <- file.path(outdir, paste0("_sb_landscapedata.csv"))
+  write.csv(data_at_gages,filepath1, row.names = FALSE)
+  filepath2 <- file.path(outdir, paste0("_sb_itemfails.csv"))
```

Comment:
Since we just created a single file, the file names are simply '_sb_landscapedata.csv' and '_sb_itemfails.csv'. I would maybe think of a simple naming convention for everything related to these targets (including function names, variable names, file names, etc.) and make sure you pick something consistent and easy to understand for everything. Keep in mind that we might call these functions later for the ungaged basins too, so it might help to indicate that these data are for the gages_ii watersheds (maybe we include 'outdir' and 'filename' both as variables up top so we can name this one 'blah_blah_gages_ii.csv' and a future one 'blah_blah_ungaged_comids.csv' or something like that.